File: micro-service-hello-server/src/main/java/demo/DemoHandler.java
package demo;

import org.springframework.web.bind.annotation.*;

@RestController
public class DemoHandler {

  @GetMapping("/get")
  public User hello1(@RequestParam("name") String name) {
    //ServiceInstance localInstance = client.getLocalServiceInstance();
    // return "Hello " + name + ": " + localInstance.getServiceId() + ":" +
    // localInstance.getHost() + ":"
    // + localInstance.getPort();
    User user = new User(name, "Hello World");
    System.out.println(user);
    return user;
  }

  @PostMapping("/post")
  public User hello2(@RequestParam("name") String name) {
    //ServiceInstance localInstance = client.getLocalServiceInstance();
    // return "Hello " + name + ": " + localInstance.getServiceId() + ":" +
    // localInstance.getHost() + ":"
    // + localInstance.getPort();
    User user = new User(name, "Hello World");
    System.out.println(user);
    return user;
  }

}

File: micro-service-hello-server/src/main/java/demo/DemoHandler.kt
package demo

import demo.bean.*
import jakarta.validation.*
import jakarta.validation.constraints.*
import org.springframework.validation.annotation.*
import org.springframework.web.bind.annotation.*


@RestController
@Validated
class DemoHandler {
  @GetMapping("/get")
  fun hello1(@RequestParam("name") name: String?): User {
    //ServiceInstance localInstance = client.getLocalServiceInstance();
    // return "Hello " + name + ": " + localInstance.getServiceId() + ":" +
    // localInstance.getHost() + ":"
    // + localInstance.getPort();
    val user = User(name, "Hello World")
    println(user)
    return user
  }

  @PostMapping("/post")
  fun hello2(@Valid user: User?, @Max(200) @NotNull @RequestParam age: Int?): User {
    //ServiceInstance localInstance = client.getLocalServiceInstance();
    // return "Hello " + name + ": " + localInstance.getServiceId() + ":" +
    // localInstance.getHost() + ":"
    // + localInstance.getPort();
    val user = User(user!!.name, "Hello World")
    println(user)
    return user
  }

}
